packages/language-server/dep-install/service.ts

```diff
@@ -0,0 +1,132 @@
+        });
+        child.stderr.on('data', chunk => {
+          stderr += chunk.toString();
+        });
+      }
+
+      child.on('exit', code => {
+        if (code !== 0) {
+          return reject({
+            code,
+            stderr
+          });
+        }
+        return resolve({
+          code,
+          stdout,
+          stderr
+        });
+      });
+    }
+  );
+}
+
+interface Options {
+  root?: string;
+  server?: IConnection;
+}
+
+export class DepInstaller {
+  private static instance: DepInstaller;
+  private root: string;
+  private server: IConnection;
+
+
+ private onRequestToInstall = async (params) => {
+    console.log('Received request to install deps:', JSON.stringify(params));
+    const DialectClass = Dialects[params.dialect];
+    if (
+      !DialectClass ||
+      !DialectClass.deps ||
+      DialectClass.deps.length === 0
+    ) {
+      throw new Error('Nothing to install. Request is invalid.');
+    }
+
+    const deps: typeof GenericDialect['deps'] = DialectClass.deps;
+
+    for (let dep of deps) {
+      switch(dep.type) {
+        case 'npmscript':
+          console.log(`Will run ${dep.name} script`);
+          await this.runNpmScript(dep.name, { env: dep.env });
+          console.log(`Finished ${dep.name} script`);
+          break;
+        case 'package':
+          console.log(`Will install ${dep.name} package`);
+          await this.install(`${dep.name}${dep.version ? `@${dep.version}` : ''}`, { env: dep.env });
+          console.log(`Finished ${dep.name} script`);
+          break;
+      }
+
+    }
+    console.log('Finished installing deps');
+  }
+
+  public boot({ server, ...options }: Options = {}): DepInstaller {
+    Object.keys(options).forEach(k => (this[k] = options[k] ));
+
+    if (!this.server && server) {
+      this.server = this.server || server;
+      this.server.onRequest(InstallDep, this.onRequestToInstall);
+    }
+
+    try {
+      fs.mkdirSync(path.join(this.root, 'node_modules'))
+    } catch (error) {}
+
+    return this;
+  }
+
+  private npm(args: ReadonlyArray<string>, options: SpawnOptions = {}) {
+    return run('npm', args, { cwd: this.root, shell: true, ...options });
+  }
+
+  get npmVersion() {
+    return this.npm(['--version']).then(({ stdout }) =>
+      stdout.replace('\n', '')
+    );
+  }
+  constructor(options: Options) {
+    if (DepInstaller.instance) {
+      DepInstaller.instance.boot(options);
+      return this;
+    }
+    DepInstaller.instance = this.boot(options);
+  }
+
+  public async install(args: string | string[], options: SpawnOptions = {}) {
+    return run('npm', ['install', ...(Array.isArray(args) ? args : [args]) ], { stdio: [ process.stdin, process.stdout, process.stderr ], ...options });
+  }
+
+  public async runNpmScript(scriptName: string, options: SpawnOptions = {}) {
+    return run('npm', ['run', scriptName ], { cwd: this.root, stdio: [ process.stdin, process.stdout, process.stderr ], ...options });
```

Comment:
```suggestion
    return run('npm', ['run', scriptName ], { cwd: this.root, stdio: [ process.stdin, process.stdout, process.stderr ], shell: true,  ...options });
```

Reply:
Adding `shell: true` does the trick: https://github.com/nodejs/node/issues/3675#issuecomment-308963807 😩